extensions/amp-access/0.1/amp-access.js

```diff
@@ -233,6 +256,108 @@ export class AccessService {
   }
 
   /**
+   * @private
+   */
+  scheduleView_() {
+    if (this.viewer_.isVisible()) {
+      this.reportWhenViewed_();
+    }
+    this.viewer_.onVisibilityChanged(() => {
+      if (this.viewer_.isVisible()) {
```

Comment:
The idea is when the document becomes visible, the async process is kicked off to see if it'd actually register a view. This is why the `isVisible` check is outside.